### PR TITLE
Update cloud databases support

### PR DESCRIPTION
**Pull request by nasark**
- also refactored spec tests

@miq-bot add_label enhancement
@miq-bot assign @agrare

### CODE REVIEW

File: app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb [hunk 1]
```diff
@@ -166,6 +167,46 @@ def self.create_mysql_database(ext_management_system, options)
...
+  def update_oracle_database(options)
+    with_provider_connection(:service => 'Database::DatabaseClient') do |connection|
+      connection.update_autonomous_database(
+        ems_ref,
+        OCI::Database::Models::UpdateAutonomousDatabaseDetails.new(
+          :display_name => options[:name]
+        )
+      )
+    end
+  end
```
Comment (nasark) [inline app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb:199]: The following settings are needed for this to work (not documented anywhere)
Screen Shot 2022-05-18 at 12 10 42 PM
k

File: spec/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database_spec.rb [hunk 2]
```diff
@@ -49,6 +45,28 @@
...
       end
     end
+
+    context '#update_oracle_database' do
+      it 'updates the Oracle database' do
+        expect(connection).to receive(:update_autonomous_database).with(
+          cloud_database.ems_ref,
+          OCI::Database::Models::UpdateAutonomousDatabaseDetails.new(:display_name => "test-db123")
+        )
+        cloud_database.db_engine = "Oracle Database"
```
[the remark this answers is not in the document]
Comment (nasark) [reply spec/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database_spec.rb:55]: I see you have grouped the update actions under a `describe "#update_database"` block, would you like me to refactor the other actions in a similar fashion? i.e. `describe #create_database` etc. At the moment I have them grouped as `describe 'oracle cloud database actions'` and `describe 'mysql cloud database actions'`

[the remark this answers is not in the document]
Comment (agrare) [reply spec/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database_spec.rb]: Not for this PR but would be a nice follow-up

[the remark this answers is not in the document]
Comment (nasark) [reply spec/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database_spec.rb]: will do, thanks

**Review by agrare — APPROVED** [review]

**Pull request merged**